src/main/java/org/springframework/integration/aws/lock/DynamoDbLockRegistry.java

```diff
@@ -279,6 +284,56 @@ public void afterPropertiesSet() {
 		this.initialized = true;
 	}
 
+	/**
+	 * Creates a DynamoDB table with the right schema for it to be used by this locking library.
+	 * The table should be set
+	 * up in advance, because it takes a few minutes for DynamoDB to provision a new instance.
+	 * <p>
+	 * This method is a variation of {@link AmazonDynamoDBLockClient#createLockTableInDynamoDB} to support custom
+	 * {@link com.amazonaws.services.dynamodbv2.model.BillingMode} for the lock table.
+	 * <p>
+	 * If table already exists no exception.
+	 */
+	private void createLockTableInDynamoDB() {
+		try {
+			KeySchemaElement partitionKeyElement = new KeySchemaElement();
+			partitionKeyElement.setAttributeName(this.partitionKey);
+			partitionKeyElement.setKeyType(KeyType.HASH);
+
+			List<KeySchemaElement> keySchema = new ArrayList<>();
+			keySchema.add(partitionKeyElement);
+
+			Collection<AttributeDefinition> attributeDefinitions = new ArrayList<>();
+			attributeDefinitions.add(new AttributeDefinition().withAttributeName(this.partitionKey)
+					.withAttributeType(ScalarAttributeType.S));
+
+			KeySchemaElement sortKeyElement = new KeySchemaElement();
+			sortKeyElement.setAttributeName(this.sortKeyName);
+			sortKeyElement.setKeyType(KeyType.RANGE);
+			keySchema.add(sortKeyElement);
+			attributeDefinitions.add(new AttributeDefinition().withAttributeName(this.sortKeyName)
+					.withAttributeType(ScalarAttributeType.S));
+
+			CreateTableRequest createTableRequest = new CreateTableRequest(this.tableName, keySchema)
+					.withAttributeDefinitions(attributeDefinitions)
+					.withBillingMode(this.billingMode.getBillingMode());
+
+			if (isProvisioned(this.billingMode)) {
+				createTableRequest.setProvisionedThroughput(
+						new ProvisionedThroughput(this.readCapacity, this.writeCapacity));
+			}
+
+			this.dynamoDB.createTable(createTableRequest);
+		}
+		catch (ResourceInUseException ex) {
+			// Swallow an exception and you should check for table existence
+		}
+	}
+
+	private static boolean isProvisioned(BillingMode billingMode1) {
+		return com.amazonaws.services.dynamodbv2.model.BillingMode.PROVISIONED.equals(billingMode1.getBillingMode());
```

Comment:
I think we won't need this method when you remove that intermediate `BillingMode` class.